### ISSUE TITLE
VOS 6.1.4: Execute DROP GRAPH / CLEAR GRAPH via public SPARQL endpoint has no effect

### ISSUE THREAD

**Issue by k00ni** [openlink/virtuoso-opensource#349]
My local machine configuration:
- Ubuntu 12.04.
- Virtuoso Server **6.1.4**

The user **SPARQL** has the following settings
- user type: WebDav und SQL/ODBC
- Account roles > selected: SPARQL_UPDATE, SPARQL_SELECT
- Primary role: administrators

I use EasyRdf's Sparql\Client to execute queries via HTTP. My SPARQL-Endpoint is Virtuoso's public http://localhost:8890/sparql.

The following statements have NO effect:
- CLEAR GRAPH http://mygraphuri/
- DROP GRAPH http://mygraphuri/
- SILENT DROP GRAPH http://mygraphuri/

The graph and all its triples remains. What did i wrong?

**Comment by indeyets** [comment]
you should wrap graph-iri like this: `CLEAR GRAPH <http://mygraphuri/>`

**Comment by rumito** [comment]
Hi,

On 20-Feb-15 11:33 AM, Konrad Abicht wrote:

> My local machine configuration:
> - Ubuntu 12.04.
> - Virtuoso Server _6.1.4_
> 
> The user _SPARQL_ has the following settings
> - user type: WebDav und SQL/ODBC
> - Account roles > selected: SPARQL_UPDATE, SPARQL_SELECT
> - Primary role: administrators
> 
> I use 
> [[EasyRdf|https://github.com/njh/easyrdf/blob/master/lib/Sparql/Client.php]] 
> to execute queries via HTTP over the public sparql-Endpoint of 
> Virtuoso: http://localhost:8890/sparql.
> 
> The following statements have NO effect:
> - CLEAR GRAPH http://mygraphuri/
> - DROP GRAPH http://mygraphuri/
> - SILENT DROP GRAPH http://mygraphuri/

If you don't know whether the graph is created explicitly or not, you 
should use:

DROP SILENT GRAPH http://mygraphuri/ ;

(not SILENT DROP GRAPH).

Best Regards,
Rumi Kocis

**Comment by k00ni** [comment]
Hey, thanks for the quick responses! 

from @indeyets:

> CLEAR GRAPH <http://mygraphuri/>

I posted it like this, but Github hide the brackets. With brackets it does not work too.

from @rumito:

> DROP SILENT GRAPH <http://mygraphuri/>;

My bad. I used both kinds, DROP SILENT GRAPH and SILENT DROP GRAPH, but same result: graph remains and kept his triples.

from @rumito:

> If you don't know whether the graph is created explicitly or not

I created the graph the same way i want to drop it, which means, i use the same function to send and receive queries.

**Comment by rumito** [comment]
Hi,

On 20-Feb-15 12:49 PM, Konrad Abicht wrote:

> Hey, thanks for the quick responses!
> 
> ```
> CLEAR GRAPH <http://mygraphuri/>
> ```
> 
> I posted it like this, but Github hide the brackets. With brackets it 
> does not work.
> 
> ```
> DROP SILENT GRAPH <http://mygraphuri/>;
> ```
> 
> My bad. I used both kinds, drop silent and silent drop, but same 
> result = graph remains and kept his triples.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/openlink/virtuoso-opensource/issues/349#issuecomment-75226657.

Hm, right, so you say you use 
[[EasyRdf|https://github.com/njh/easyrdf/blob/master/lib/Sparql/Client.php]] 
to execute the statements from above.

What about if you execute them from iSQL or from host:port/conductor, 
would then these have effect i.e. execute them not as user SPARQL but 
rather as user dba?

Best Regards,
Rumi Kocis

**Comment by k00ni** [comment]
> What about if you execute them from iSQL or from host:port/conductor,
> would then these have effect i.e. execute them not as user SPARQL but
> rather as user dba?

I am not sure how to authenticate on the Virtuoso server via HTTP. Do you mean login into Virtuoso using HTTP requests and send queries after that?

Login into the conductor with dba/dba and executing the following queries over http://localhost:8890/sparql results in:

#### DROP GRAPH <http://localhost/Saft/TestGraph/>

```
Virtuoso 22023 Error SPARUL_DROP() failed: 
graph <http://localhost/Saft/TestGraph/> has not been explicitly created before

SPARQL query:
define sql:big-data-const 0 
#output-format:text/html
DROP GRAPH <http://localhost/Saft/TestGraph/>
```

Graph remains and kept his triples.

#### DROP SILENT GRAPH <http://localhost/Saft/TestGraph/>

```
callret-0
Drop silent graph <http://localhost/Saft/TestGraph/> -- 
graph has not been explicitly created before, triples were removed
```

**BUT** Graph was removed **successfully**.

**Comment by rumito** [comment]
On 20-Feb-15 1:39 PM, Konrad Abicht wrote:

> Running it via conductor's web interface 
> (http://localhost:8890/sparql) returns following results:

The Conductor UI is accessible from  http://localhost:8890/conductor 
however. There you need to log in as user dba.
The sparql user may not have enough rights so to drop graphs from the 
sparql endpoint http://localhost:8890/sparql ..

What happens if you:

1) drop the graph from the Conductor UI
2) drop the graph from iSQL tool i.e: isql localhost:{port} dba {dba-pwd}  ?

**Comment by k00ni** [comment]
For me **localhost:8890/sparql** and **localhost:8890/conductor** were the same. :)

#### Conductor test

I logged in localhost:8890/conductor as dba/dba
Dropping a graph via conductor > Linked Data > Graphs > Delete Graph works like a charm.

#### iSQL test

I logged in via **isql VOS dba dba**

Executing **DROP SILENT GRAPH <http://localhost/Saft/TestGraph/>** results in an error:
`[ISQL]ERROR: Could not SQLPrepare`

Same error for **DROP GRAPH <http://localhost/Saft/TestGraph/>**.

**Comment by k00ni** [comment]
**For your information:** Working with Virtuoso via PHP + PDO/ODBC and dba/dba works without any problems.

**Comment by k00ni** [comment]
A message at the debian mailing list, https://lists.debian.org/debian-qt-kde/2013/08/msg00020.html, about the release of Virtuoso 7.0 contains:

> Added support for SPARQL 1.1 Update Protocol

And the site http://virtuoso.openlinksw.com/whats-new/ from Virtuoso about the version history of the server said for Virtuoso **6.1**:

> SPARQL 1.1 syntax for INSERT DATA / DELETE DATA
> SPARQL 1.1 HAVING clause for filtering on GROUP BY

Better SPARQL 1.1 support starts with **6.4**.

I guess Virtuoso 6.1.x just lacks the ability to process SPARQL UPDATE queries via HTTP?

**Comment by HughWilliams** [comment]
SPARQL 1.1 support was added in 6.1.5+ and you indicate running 6.1.4 , thus you should upgrade to the latest 6.1.8 release, see:

http://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VOSNews#2013-12-10%20--%20Virtuoso%20Open-Source%20Edition%206.1.8%20Released

Or even the latest 7.2.0 release , which we would always recommend, but may take more time to certify your application against ...

**Comment by k00ni** [comment]
Hi Hugh,

i struggled finding deb-packages for Virtuoso 6.1.8 or higher. Downloading via http://stack.linkeddata.org/download/repo.php and http://stack.linkeddata.org/download-and-install/ does not really work and resulted in: 

```
Ign http://stack.linkeddata.org ldstack-virt6/main TranslationIndex
Err http://stack.linkeddata.org ldstack-virt6/main i386 Packages
  404  Not Found
Ign http://stack.linkeddata.org ldstack/main Translation-en_US
Ign http://stack.linkeddata.org ldstack/main Translation-en
Ign http://stack.linkeddata.org ldstack-virt6/main Translation-en_US
Ign http://stack.linkeddata.org ldstack-virt6/main Translation-en
Fetched 4,166 B in 6s (675 B/s)                                                                                                                       
W: Failed to fetch http://stack.linkeddata.org/deb/dists/ldstack-virt6/main/binary-i386/Packages  404  Not Found

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

Anyway. I think this issue can be closed. As @HughWilliams wrote, at least Virtuoso 6.1.8 supports SPARQL UPDATE via HTTP.

**Comment by HughWilliams** [comment]
The lod2 stack debian packages stopped at 6.1.4 as then moved to 7.0 when it was released and the last version was 7.1  ...

So you would have to compile the 6.1.8 from source if you want to use it or there is a 7.1 debian package in the lod2 stack repo ...
